index.js

```diff
@@ -194,7 +194,7 @@ class KeyringController extends EventEmitter {
   // All Keyring classes implement a unique `type` string,
   // and this is used to retrieve them from the keyringTypes array.
   async addNewKeyring (type, opts) {
-    if (opts.password) {
+    if (opts && opts.password) {
```

Comment:
mind adding a quick test for this in test/index.js?

Reply:
@diracdeltas done. It was a little more challenging than expected because the two keyrings that are included by default actually need a certain things passed into `opts`. I created a mock keyring that doesn't need `opts`, and just for the one test, a new `KeyringController` that includes that keyring type.